BigBaseV2/src/script.hpp

```diff
@@ -8,20 +8,24 @@ namespace big
 	public:
 		using func_t = void(*)();
 	public:
-		explicit script(func_t func, std::optional<std::size_t> stack_size = std::nullopt);
+		explicit script(func_t func, const char* name, bool* enabled, std::optional<std::size_t> stack_size = std::nullopt);
 		~script();
 
 		void tick();
 		void yield(std::optional<std::chrono::high_resolution_clock::duration> time = std::nullopt);
 		static script* get_current();
 		static void script_exception_handler(PEXCEPTION_POINTERS exp);
+		const char* m_name;
```

Comment:
Use `const std::string_view`, string view offers more flexibility as you can pass `std::string` and `char[]` interchangeably it's also more efficient than `std::string` as string view doesn't make a copy of the value but instead references the original storage location of the string.